backends/mongo.go

```diff
@@ -108,7 +108,13 @@ func NewMongo(authOpts map[string]string, logLevel log.Level, hasher hashing.Has
 		m.insecureSkipVerify = true
 	}
 
-	addr := fmt.Sprintf("mongodb://%s:%s", m.Host, m.Port)
+	var addr string
+
+	if mongoUri, ok := authOpts["mongo_uri"]; ok {
+		addr = mongoUri
```

Comment:
Is there some standard validation to run against the `uri`?

Reply:
The existing `mongo.Connect` method will validate it and return an appropriate error if the URI is bad.